gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -120,6 +130,13 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
     viewModel.getShowAddFeatureDialogRequests().observe(this, this::onShowAddFeatureDialogRequest);
     viewModel.getFeatureSheetState().observe(this, this::onFeatureSheetStateChange);
     viewModel.getOpenDrawerRequests().observe(this, __ -> openDrawer());
+
+    showFeatureDialogRequests = PublishSubject.create();
+
+    showFeatureDialogRequests
+        .flatMapMaybe(__ -> addFeatureDialogFragment.show(getChildFragmentManager()))
```

Comment:
`switchMapMaybe()`?

Reply:
Note to self: The stream returned by show() should never return an error, but if it did this would break. In the future let's always make the expectations for returned Rx streams clear in Javadoc (i.e. errors should be handled, or never should fail).

Reply:
Sounds good! We could perhaps even write a simple annotation for communicating/documenting this--actually I don't know a lot about annotations, but It'd be cool if we could even implement a type annotation for error-prone streams, similar to `@NonNull` and company. 

We can use `switchMapMaybe` but we need to upgrade the RX version to `2.2.8`. Right now, if I attempt that upgrade, I run into the following exception:

```
2019-04-29 12:13:54.251 5257-5257/com.google.android.gnd E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.google.android.gnd, PID: 5257
    java.lang.NoSuchFieldError: No field actual of type Lio/reactivex/Observer; in class Lhu/akarnokd/rxjava2/debug/ObservableOnAssembly$OnAssemblyObserver; or its superclasses (declaration of 'hu.akarnokd.rxjava2.debug.ObservableOnAssembly$OnAssemblyObserver' appears in /data/app/com.google.android.gnd-8_H90TGFn6liYW29D3xoMQ==/split_lib_dependencies_apk.apk!classes2.dex)
        at hu.akarnokd.rxjava2.debug.ObservableOnAssembly$OnAssemblyObserver.onNext(ObservableOnAssembly.java:55)
        at io.reactivex.internal.operators.observable.ObservableSkip$SkipObserver.onNext(ObservableSkip.java:56)
        at com.uber.autodispose.android.lifecycle.LifecycleEventsObservable$ArchLifecycleObserver.onStateChange(LifecycleEventsObservable.java:117)
        at com.uber.autodispose.android.lifecycle.LifecycleEventsObservable_ArchLifecycleObserver_LifecycleAdapter.callMethods(LifecycleEventsObservable_ArchLifecycleObserver_LifecycleAdapter.java:22)
        at androidx.lifecycle.SingleGeneratedAdapterObserver.onStateChanged(SingleGeneratedAdapterObserver.java:36)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:355)
        at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:293)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:333)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:138)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:124)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:2485)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1494)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1852)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3269)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStart(FragmentManager.java:3235)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:2483)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1494)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1852)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3269)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStart(FragmentManager.java:3235)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:2483)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1494)
        at androidx.fragment.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1852)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3269)
        at androidx.fragment.app.FragmentManagerImpl.dispatchStart(FragmentManager.java:3235)
        at androidx.fragment.app.FragmentController.dispatchStart(FragmentController.java:212)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:628)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:178)
        at com.google.android.gnd.MainActivity.onStart(MainActivity.java:140)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391)
        at android.app.Activity.performStart(Activity.java:7157)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2937)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
2019-04-29 12:13:54.252 5257-5257/com.google.android.gnd E/AndroidRuntime:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

Reply:
Looking at the stack trace it appears to be happening in AutoDispose, which is probably expecting an older version. Have you tried updating the autodispose dep to the latest as well? ([1.2.0](https://mvnrepository.com/artifact/com.uber.autodispose/autodispose)?)

Reply:
Turns out RxJavaDebug was the culprit. I've upgraded that, RxJava, and Autodispose.